scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -30,9 +39,12 @@ class CoverageRunner(
    */
   fun runWithCoverageAsync(
     bazelTestTarget: String
-  ): Deferred<List<String>?> {
+  ): Deferred<CoverageReport> {
     return CoroutineScope(scriptBgDispatcher).async {
-      retrieveCoverageResult(bazelTestTarget)
+      val coverageResult = retrieveCoverageResult(bazelTestTarget)
+        ?: throw RuntimeException("Failed to retrieve coverage result for $bazelTestTarget")
```

Comment:
Use ``error`` rather than a generic ``RuntimeException``--it throws an ``IllegalStateException`` which is contractually more appropriate for the failure being caught here.

Reply:
Done used `error` accordingly!